src/main/scala/leon/evaluators/RecursiveEvaluator.scala

```diff
@@ -231,7 +232,47 @@ abstract class RecursiveEvaluator(ctx: LeonContext, prog: Program, maxSteps: Int
 
     case RealMinus(l,r) =>
       e(RealPlus(l, RealUMinus(r)))
-
+      
+    case StringConcat(l, r) =>
+      (e(l), e(r)) match {
+        case (StringLiteral(i1), StringLiteral(i2)) => StringLiteral(i1 + i2)
+        case (le,re) => throw EvalError(typeErrorMsg(le, StringType))
+      }
+    case StringLength(a) => e(a) match {
+      case StringLiteral(a) => InfiniteIntegerLiteral(a.length)
+      case res => throw EvalError(typeErrorMsg(res, IntegerType))
+    }
+    case SubString(a, start, end) => (e(a), e(start), e(end)) match {
+      case (StringLiteral(a), InfiniteIntegerLiteral(b), InfiniteIntegerLiteral(c))  =>
+        StringLiteral(a.substring(b.toInt, c.toInt))
```

Comment:
There is a disconnect between what you have here and what you implement in runtime.StringOps.

in runtime.StringOps you check for erroneous conditions, and return "" in these cases, in this evaluator  it will throw an exception which should turn into a RuntimeError, which is the right thing.